src/Init/System/IO.lean

```diff
@@ -506,7 +507,7 @@ structure Child (cfg : StdioConfig) where
 
 @[extern "lean_io_process_spawn"] constant spawn (args : SpawnArgs) : IO (Child args.toStdioConfig)
 
-@[extern "lean_io_process_child_wait"] constant Child.wait {cfg : @& StdioConfig} : @& Child cfg → IO UInt32
+@[extern "lean_io_process_child_wait"] constant Child.wait {cfg : @& StdioConfig} : @& Child cfg → BaseIO UInt32
```

Comment:
`waitpid` can return an error though?  (We also don't seem ensure that `Child.wait` is called exactly once.)

Reply:
@gebner Lean folds the error from `waitpid` into the exit code. As such, `Child.wait` cannot error.

Your comment also demonstrates one of the reasons why using `BaseIO` is so important. Without it, the user has to go look at the internal implementation to figure out whether something can actually error (as whether it can is not always intuitive).

Reply:
Gabriel is right, we should not ignore the return value of `waitpid`. Waiting twice should fail.

Reply:
@Kha do you want push a change to update the implementation of `lean_io_process_child_wiat` and then I can update this PR accordingly?

Reply:
done